src/java.base/share/classes/com/sun/crypto/provider/GaloisCounterMode.java

```diff
@@ -497,34 +497,75 @@ int encrypt(byte[] in, int inOfs, int len) {
      * @param len the length of the input data
      * @param out the buffer for the result
      * @param outOfs the offset in <code>out</code>
-     * @exception ProviderException if <code>len</code> is not
-     * a multiple of the block size
      * @return the number of bytes placed into the <code>out</code> buffer
      */
-    int encrypt(byte[] in, int inOfs, int len, byte[] out, int outOfs) {
-        checkDataLength(processed, len);
+    int encrypt(byte[] in, int inOfs, int inLen, byte[] out, int outOfs) {
+        checkDataLength(processed, inLen);
 
         processAAD();
-        int remainder = len % blockSize;
-        if ( remainder > 0) {
+        // 'len' stores the length to use with buffer 'in'.
+        // 'inLen' stores the length returned by the method.
```

Comment:
line 551 has "return len;" which seems conflicting with these two-line comments here?

Reply:
The comment is backwards.  I wrote the comment before I decided to switch the meaning of the variables and never updated the comment